osc-server/src/main/java/org/osc/core/broker/service/securitygroup/DeleteSecurityGroupService.java

```diff
@@ -64,6 +64,9 @@ public BaseJobResponse exec(BaseDeleteRequest request, EntityManager em) throws
         try {
             unlockTask = LockUtil.tryLockSecurityGroup(securityGroup,
                     securityGroup.getVirtualizationConnector());
+            //remove sfc link to SG
+            securityGroup.setServiceFunctionChain(null);
```

Comment:
this has no effect on the force delete since this transaction is not committed before the job starts...

Assuming force delete happens after the regular delete, it might be fine.